### PR TITLE
New test setup

### PR DESCRIPTION
**Pull request by Zifzaf**
Check the following:

 - [x] Breaking changes marked in commit message
 - [na] Changelog updated
 - [x] Code style cleaned (ran `make style-fix`)
 - [x] Tested on actual hardware

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,6 @@
 /sps30-i2c/sps30_example_usage
 /sps30-i2c/user_config.inc
 /sps-common/sps_git_version.c
+i2c-mux-testbed/
+	tests/sps30-test-hw_i2c
+	tests/sps30-test-sw_i2c
```
Comment (rnestler) [inline .gitignore:8]: Something messed up with the whitespace?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -1,20 +1,34 @@
...
-    --form ref=master
-    --form token=$CI_JOB_TOKEN
-    --form "variables[DRIVER_NAMES]=embedded-sps"
-    --form "variables[DRIVER_BRANCH]=$CI_COMMIT_REF_NAME"
-    --form "variables[DRIVER_COMMIT_SHA]=$CI_COMMIT_SHA"
-    https://gitlab/api/v4/projects/801/trigger/pipeline
+    - git clone -b 2.0.0 https://gitlab-ci-token:${CI_JOB_TOKEN}@gitlab/MSO-SW/drivers/testbed/i2c-mux-testbed.git
+    - cd tests
+    - make test
+
```
Comment (rnestler) [inline .gitlab-ci.yml:34]: Trailing newline.

**Review by rnestler — COMMENTED** [review]
Just some minor nitpicks.

File: .gitignore (hunk 1 above)
Comment (abrauchli) [reply .gitignore]: / prefix for all?

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
  * [`changed`] Update timings according to datasheet
  * [`changed`] Use configuration independent endianness conversions: no need to
                define `SENSIRION_BIG_ENDIAN` anymore.
+ * [`changed`] Makefile to only include needed files from embedded-common
```
Comment (rnestler) [inline CHANGELOG.md:12]: This looks unrelated. Probably not rebased correctly?

File: tests/sps30-test.cpp [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+        sps30_test_setup();
+    }
+
+    void teardown() {
+        sps30_delayed_reset();
+        sensirion_i2c_release();
+    }
+};
+
+TEST (SPSTestGroup, SPS30Test_no_cleaning) { sps30_test(0); }
```
Comment (rnestler) [inline tests/sps30-test.cpp:149]: These look strange. Is it possible to configure clang-format to avoid this?

**Review by rnestler — CHANGES_REQUESTED** [review]
Seems to need rebasing and maybe we can fix the clang-format issue for single line tests as well.

File: tests/sps30-test.cpp (hunk 4 above)
Comment (rnestler) [reply tests/sps30-test.cpp:149]: Resolved offline: It is to cumbersome to get clang-format to behave correctly here. It seems to be a bug in clang-format. See https://stackoverflow.com/questions/51535549/format-foreach-macros-the-same-as-loops-with-clang-format

**Review by rnestler — APPROVED** [review]
I rebased and edited the commit messages to explain a bit better what is going on in each commit.

**Review by abrauchli — APPROVED** [review]
